test/experimental/test_vocab.py

```diff
@@ -0,0 +1,209 @@
+# -*- coding: utf-8 -*-
+from collections import OrderedDict
+import os
+import torch
+
+from test.common.torchtext_test_case import TorchtextTestCase
+from torchtext.experimental.vocab import (
+    Vocab
+)
+
+
+class TestVocab(TorchtextTestCase):
+    def tearDown(self):
+        super().tearDown()
+        torch._C._jit_clear_class_registry()
+        torch.jit._recursive.concrete_type_store = torch.jit._recursive.ConcreteTypeStore()
+
+    def test_has_unk(self):
+        c = OrderedDict({})
+        v = Vocab(c)
+
+        # check if unk is mapped to the first index
+        self.assertEqual(v['not_in_it'], 0)
+        self.assertEqual(v['<unk>'], 0)
+
+    def test_new_unk(self):
+        c = OrderedDict({})
+        v = Vocab(c, specials=('<new_unk>',), unk_token="<new_unk>")
+
+        # check if new_unk is mapped to the first index
+        self.assertEqual(v['<new_unk>'], 0)
+        self.assertEqual(v['not_in_it'], 0)
+
+    def test_vocab_get_item(self):
+        token_to_freq = {'a': 2, 'b': 2}
+        sorted_by_freq_tuples = sorted(token_to_freq.items(), key=lambda x: x[1], reverse=True)
+        c = OrderedDict(sorted_by_freq_tuples)
+        v = Vocab(c)
+
+        self.assertEqual(v['<unk>'], 0)
+        self.assertEqual(v['<pad>'], 1)
+        self.assertEqual(v['a'], 2)
+        self.assertEqual(v['b'], 3)
+
+    def test_vocab_set_item(self):
+        c = OrderedDict({'a': 2})
+
+        # add item to end
+        v = Vocab(c)
+        v.insert_token('b', 3)
+
+        self.assertEqual(v['<unk>'], 0)
+        self.assertEqual(v['<pad>'], 1)
+        self.assertEqual(v['a'], 2)
+        self.assertEqual(v['b'], 3)
+
+        # add item to middle
+        v = Vocab(c, specials_first=False)
+        v.insert_token('b', 0)
+
+        self.assertEqual(v['b'], 0)
+        self.assertEqual(v['a'], 1)
+        self.assertEqual(v['<unk>'], 2)
+        self.assertEqual(v['<pad>'], 3)
+
+    def test_vocab_append_token(self):
+        c = OrderedDict({'a': 2})
+        v = Vocab(c)
+        v.append_token('b')
+
+        self.assertEqual(len(v), 4)
+        self.assertEqual(v['b'], 3)
+
+    def test_vocab_len(self):
+        token_to_freq = {'a': 2, 'b': 2, 'c': 2}
+        sorted_by_freq_tuples = sorted(token_to_freq.items(), key=lambda x: x[1], reverse=True)
+        c = OrderedDict(sorted_by_freq_tuples)
+        v = Vocab(c)
+
+        self.assertEqual(len(v), 5)
+
+    def test_vocab_basic(self):
+        token_to_freq = {'hello': 4, 'world': 3, 'ᑌᑎIᑕOᗪᕮ_Tᕮ᙭T': 5, 'freq_too_low': 2}
```

Comment:
I might miss it. Do we have a test case starting from a list of tokens?

Reply:
What do you mean by starting off with a list of tokens? Because the `Vocab` class expects an `OrderedDict` as the input. So even if we did start off with a list of tokens, we would have to probably get a dictionary of tokens to frequency and possibly sort this dictionary in a similar manner that I am doing now to get an ordered list of tuples which are finally fed into an OrderedDict class.

Does this make sense?